clients/fides-js/src/fides.ts

```diff
@@ -120,15 +182,58 @@ const init = async ({
   geolocation,
   options,
 }: FidesConfig) => {
+  const { fides_user_device_id: fidesUserDeviceId } =
+    getOrMakeFidesCookie().identity;
+
+  if (!validateOptions(options)) {
+    debugLog(options.debug, "Invalid overlay options", options);
+    return;
+  }
+
+  const fidesRegionString = await retrieveEffectiveRegionString(
+    geolocation,
+    options
+  );
+  let effectiveExperience: PrivacyExperience | undefined | null = experience;
+  let shouldInitOverlay: boolean = !options.isOverlayDisabled;
+
+  if (!fidesRegionString) {
+    debugLog(
+      options.debug,
+      `User location could not be obtained. Skipping overlay initialization.`
+    );
+    shouldInitOverlay = false;
+  } else if (!effectiveExperience) {
+    effectiveExperience = await fetchExperience(
+      fidesRegionString,
+      options.fidesApiUrl,
+      fidesUserDeviceId,
+      options.debug
+    );
+  }
+
+  if (shouldInitOverlay && experienceIsValid(effectiveExperience, options)) {
+    await initOverlay(<OverlayProps>{
+      experience: effectiveExperience,
+      fidesRegionString,
+      options,
+    }).catch(() => {});
+  }
+
   // Configure the default consent values
   const context = getConsentContext();
   const consentDefaults = makeConsentDefaults({
+    experience,
```

Comment:
I understand why you've done this, to mirror the existing flow, but to me this smells feels really "late" to be setting defaults on the cookie - after all, those defaults should also be what's driving the look & feel for the banner.

I'd recommend taking another pass at this without editing the `makeConsentDefaults` function at all, and instead just modifying `cookie.consent` with the defaults provided from the experience data - it just feels less likely to result in an unexpected issue here...

Reply:
Updated to only `getOrMakeFidesCookie` once with the legacy values, then overwriting `cookie.consent` if we have new experiences.